tern/utils/rootfs.py

```diff
@@ -86,8 +86,8 @@ def set_up():
 def extract_layer_tar(layer_tar_path, directory_path):
     '''Assuming all the metadata for an image has been extracted into the
     temp folder, extract the tarfile into the required directory'''
-    with tarfile.open(layer_tar_path) as tar:
-        tar.extractall(directory_path)
+    os.makedirs(directory_path)
+    root_command(extract_tar, layer_tar_path, '-C', directory_path)
 
```

Comment:
Good catch! The trouble with shelling out to the system's tar utility is that it won't check for empty tarballs as we have seen in #430. I wonder if we can use `tarfile` to see if the tarball is empty or not before shelling out to it. If you can do this, I can close #226 as this change will resolve that one as well.

Reply:
Yes. I will check for empty tarballs.

Reply:
@nishakm tarfile seems to not help. But `tar -tf` will help in this case. 
But bandit is throwing some security issues. Any clue on how to fix them ?

Reply:
@nishakm `tarfile` module is not helping. `tar -tf` helps.  But there are some security warnings associated with this. Any clue how to resolve them ?

If they can't be resolved then we may have to find another way to validate tar files.

Reply:
@PrajwalM2212 do you get any warnings when running `tar -tf` on an ubuntu image? I don't see any.